src/app/components/LongDataDisplay/index.tsx

```diff
@@ -1,47 +1,71 @@
-import Collapse from '@mui/material/Collapse'
-import Link from '@mui/material/Link'
-import { FC, useState } from 'react'
+import { FC, useEffect, useRef, useState } from 'react'
 import { useTranslation } from 'react-i18next'
+import Button from '@mui/material/Button'
 import Typography from '@mui/material/Typography'
+import { styled } from '@mui/material/styles'
+import { COLORS } from '../../../styles/theme/colors'
 
-export const LongDataDisplay: FC<{ data: string; threshold: number; fontWeight?: number }> = ({
+const StyledButton = styled(Button)(() => ({
+  color: COLORS.brandDark,
+  fontWeight: 700,
+  minWidth: 'auto',
+  paddingLeft: 0,
+  '&&:hover': {
+    textDecoration: 'none',
+    color: COLORS.brandDark,
+  },
+}))
+
+const lineHeight = 22
+
+export const LongDataDisplay: FC<{ data: string; fontWeight?: number; collapsedLinesNumber?: number }> = ({
   data,
-  threshold,
   fontWeight = 700,
+  collapsedLinesNumber = 2,
 }) => {
   const { t } = useTranslation()
-  const [showData, setShowData] = useState(false)
-  const needsHiding = data.length > threshold
-  if (!needsHiding) {
-    return (
+  const [isExpanded, setIsExpanded] = useState(false)
+  const [isOverflowing, setIsOverflowing] = useState(false)
+  const textRef = useRef<HTMLDivElement | null>(null)
+  const collapsedContainerMaxHeight = collapsedLinesNumber * lineHeight
+
+  useEffect(() => {
+    const checkOverflow = () => {
+      if (textRef.current) {
+        const isOverflow = textRef.current.scrollHeight > textRef.current.clientHeight
+        setIsOverflowing(isOverflow)
+      }
+    }
+
+    checkOverflow()
+    const handleResize = () => {
+      checkOverflow()
+    }
+
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+  }, [data])
+
+  return (
+    <div>
       <Typography
-        variant="mono"
-        sx={{
+        ref={textRef}
+        style={{
           fontWeight,
+          maxHeight: isExpanded ? 'none' : collapsedContainerMaxHeight,
+          overflow: 'hidden',
+          display: 'flex',
+          lineClamp: isExpanded ? 'none' : collapsedLinesNumber,
```

Comment:
This line is completely ignored. If you can use
```css
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;  
  overflow: hidden;
```
it should work and show `...` at end

Reply:
Aw do we not want `...`?

Reply:
with show button it's clear that data is trimmed. will ask Don